x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields_button.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo, useState, useCallback } from 'react';
+import type { FC } from 'react';
+import { EuiButtonEmpty, EuiToolTip } from '@elastic/eui';
+import type { TimelineEventsDetailsItem } from '@kbn/timelines-plugin/common';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { RuleResponse } from '../../../../../common/api/detection_engine';
+import { getDefineStepsData } from '../../../../detection_engine/common/helpers';
+import { useRuleIndexPattern } from '../../../../detection_engine/rule_creation_ui/pages/form';
+import { useDefaultIndexPattern } from '../../../../detection_engine/rule_management/hooks/use_default_index_pattern';
+import { useHighlightedFieldsPrivilege } from '../../shared/hooks/use_highlighted_fields_privilege';
+import { HighlighedFieldsModal } from './highlighted_fields_modal';
+import {
+  HIGHLIGHTED_FIELDS_EDIT_BUTTON_TEST_ID,
+  HIGHLIGHTED_FIELDS_EDIT_BUTTON_TOOLTIP_TEST_ID,
+} from './test_ids';
+
+interface EditHighlighedFieldsButtonProps {
+  /**
+   * The default highlighted fields
+   */
+  customHighlightedFields: string[];
+  /**
+   * The data formatted for field browser
+   */
+  dataFormattedForFieldBrowser: TimelineEventsDetailsItem[];
+  /**
+   * The rule
+   */
+  rule: RuleResponse;
+  /**
+   * Whether the rule exists
+   */
```

Comment:
Would you be ok improving this documentation? It confused me at first. Then I looked at where we get that information from and this is what I found from the original hook returning the data

```
We try to fetch the rule first. If the request fails with 404, that could mean that the rule was deleted.
In that case, try to fetch the latest alert generated by the rule and retrieve the rule data from the alert (fallback).
```

So if I understand correctly, we could have a rule data but the rule could have been deleted, correct?